src/app/components/sidenav/sidenav.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Icon } from 'src/app/models/icon';
+
+enum Role{
```

Comment:
Move the enum into a separate file where you keep all constants and enums you use across the entire client app.